extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -145,10 +150,11 @@ export class PublishOptionsDialog {
 		// Set the publish deploymentoptions with the updated table component values
 		this.publish.setDeploymentOptions(this.optionsModel.deploymentOptions);
 		this.disposeListeners();
-
 		if (this.optionsChanged) {
 			TelemetryReporter.sendActionEvent(TelemetryViews.PublishOptionsDialog, TelemetryActions.optionsChanged);
 		}
+		// When options are Reset to default and clicked Ok, seting optionsChanged flag to false, if not set the state of the option change
+		this.publish.publishOptionsModified = this.isResetOptionsClicked ? false : (this.optionsChanged || this.publish.publishOptionsModified);
```

Comment:
what if the user changed an option or two after clicking reset?

Reply:
Updated the total scenarios in the description, let me know if you think of any other scenarios. Thanks!